### PR TITLE
[PLAY-1506] Fix `@import playbook.scss` Errors - Fix Theming / Overriding SCSS Variables (e.g., $z_10)

### PR DESCRIPTION
**Pull request by kangaree**
**What does this PR do?**

- ✅ Fix `@import playbook.scss` errors. Allow theming (e.g., set $z_10).
- ✅ Remove [PLAY-1414](https://github.com/powerhome/playbook/pull/3520) `calc()` functions that causes undefined variable errors.
- ✅ Simplify by using standard SCSS calculations instead of calc() throughout playbook
- ✅ Copy and use intl-tel-input node_modules CSS file so it works with SCSS.
- ✅ Update "how to theme" doc
- ✅ Prepare rails-react-example for PB 14.0.0 upgrade (add nullish coalescing operator to babel config)

**How to test?** Steps to confirm the desired behavior:
1. Go to /kits page
2. Review and look for major style breaking changes

#### Checklist:
- [x] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [x] **DEPLOY** I have added the `milano` label to show I'm ready for a review.

### CODE REVIEW

**Comment by kangaree** [comment]
This was only used for investigation. No changes are necessary in Playbook for now.

File: examples/rails-react-example/babel.config.js [hunk 1]
```diff
@@ -46,6 +46,7 @@ module.exports = function(api) {
       ]
     ].filter(Boolean),
     plugins: [
+      '@babel/plugin-proposal-nullish-coalescing-operator',
```
Comment (kangaree) [inline examples/rails-react-example/babel.config.js:49]: When you bump playbook to 14.0.0 in rails-react-example, you get this error:

```Module parse failed: Unexpected token (27046:173)
File was processed with these loaders:
 * ./node_modules/babel-loader/lib/index.js
You may need an additional loader to handle the result of these loaders.
|     null == h2 ? g2 = true : 0 === h2.length ? g2 = false : (g2 = Ca.exec(h2), h2 = Da.exec(h2), g2 = null !== g2 || null !== h2);
```

File: playbook/app/pb_kits/playbook/pb_avatar_action_button/_avatar_action_button.scss [hunk 2]
```diff
@@ -35,8 +35,8 @@
     &[class*=_#{$name}] {
       
       $avatar_size: map-get($avatar-sizes, $name);
-      $icon_size: calc($avatar_size / 2);
-      $border_size: calc($icon_size / 10);
+      $icon_size: $avatar_size / 2;
+      $border_size: $icon_size / 10;
```
Comment (kangaree) [inline playbook/app/pb_kits/playbook/pb_avatar_action_button/_avatar_action_button.scss:39]: Without the removal of calc() in setting this scss variable, errors in the build show:
https://ci.powerapp.cloud/blue/organizations/jenkins/powerhome%2Fnitro-web/detail/PR-41983/12/pipeline/

I think the combination of map-get and calc don't jive well?

File: playbook/app/pb_kits/playbook/pb_circle_icon_button/_circle_icon_button.scss [hunk 3]
```diff
@@ -16,7 +16,7 @@ $pb_button_styles: (
   flex-grow: 0;
   width: $pb_button_size;
   height: $pb_button_size;
-  border-radius: calc($pb_button_size/2);
+  border-radius: $pb_button_size / 2;
```
Comment (kangaree) [inline playbook/app/pb_kits/playbook/pb_circle_icon_button/_circle_icon_button.scss:19]: Calculate in SCSS and not with calc() to fix border-radius.

File: playbook/app/pb_kits/playbook/pb_avatar_action_button/_avatar_action_button.scss (hunk 2 above)
Comment (kangaree) [reply playbook/app/pb_kits/playbook/pb_avatar_action_button/_avatar_action_button.scss:39]: We have other places that use `calc()` and scss variables, but there's string interpolation`#{}`:

`height: calc(100% - #{$space-xxs * 2});`

**Review by nidaqg — APPROVED** [review]

**Review by jasperfurniss — APPROVED** [review]

**Pull request merged**